op-challenger/fault/solver/solver.go

```diff
@@ -46,6 +46,17 @@ func (s *Solver) NextMove(claim types.Claim, agreeWithClaimLevel bool) (*types.C
 	}
 }
 
+// OracleData returns the pre-image data to populate the oracle with
+// if any exists.
+func (s *Solver) OracleData(claim types.Claim) (types.PreimageOracleData, error) {
```

Comment:
I suspect this should just be provided as part of `StepData` returned from `AttemptStep`. You only ever need it when you need to perform a step.

And then the trace provider could probably return it as part of `GetPreimage` (it may be worth introducing a struct there rather than just adding more and more return values mind you.)